
syntax = "proto3";

package Agent;

service Agent {
  rpc RpcAgentStream (AgentRequest) returns (stream AgentResponse) {}
  rpc RpcAudioAndLip(AudioAndLipRequest) returns (AudioAndLipResponse);
  rpc RpcAddConversation(AddConversationRequest) returns (AddConversationResponse);
  rpc RpcUpdateConversationTitle(UpdateConversationTitleRequest) returns (UpdateConversationTitleResponse);
  rpc RpcGetConversations(GetConversationsRequest) returns (GetConversationsResponse);
  rpc RpcDeleteConversation(DeleteConversationRequest) returns (DeleteConversationResponse);
  rpc RpcGetLastMsgId(GetLastMsgIdRequest) returns (GetLastMsgIdResponse);
  rpc RpcGetMessages(GetMessagesRequest) returns (GetMessagesResponse);
  rpc RpcAliyunSts(AliyunStsRequest) returns (AliyunStsResponse);
  rpc RpcAudioToText(AudioToTextRequest) returns (AudioToTextResponse);
}

message AgentRequest {
  int64 user_id = 1;
  int64 conversation_id = 2;
  string content = 3;
}

message AgentResponse {
  string text = 1;
}

message AudioAndLipRequest {
  string text = 1;
  string model = 2;
  string voice = 3;
  float speed = 4;
}

message AudioAndLipResponse {
  string audio_file = 1;
  string lips_data = 2;
}

message AddConversationRequest {
  string user_id = 1;
}

message AddConversationResponse {
  string conversation_id = 1;
}

message UpdateConversationTitleRequest {
  string user_id = 1;
  string conversation_id = 2;
  string title = 3;
}

message UpdateConversationTitleResponse {
  bool is_success = 1;
}

message GetConversationsRequest {
  string user_id = 1;
}

message GetConversationsResponse {
  repeated string conversations = 1;
}

message DeleteConversationRequest {
  string user_id = 1;
  string conversation_id = 2;
}

message DeleteConversationResponse {
  bool is_success = 1;
}

message GetLastMsgIdRequest {
  string user_id = 1;
  string conversation_id = 2;
}

message GetLastMsgIdResponse {
  string msg_id = 1;
}

message GetMessagesRequest {
  string user_id = 1;
  string conversation_id = 2;
}

message GetMessagesResponse {
  repeated string messages = 1;
}

message AliyunStsRequest {
  string user_id = 1;
}

message AliyunStsResponse {
  string access_key_id = 1;
  string access_key_secret = 2;
  string expiration = 3;
  string security_token = 4;
}

message AudioToTextRequest {
  string user_id = 1;
  bytes content = 2;
}

message AudioToTextResponse {
  string text = 1;
}

